### PR TITLE
Basin hopping continued

### PR DESCRIPTION
**Pull request by Martin092**
Working implementation of basin hopping.

### CODE REVIEW

File: Disturber.py [hunk 1]
```diff
@@ -129,4 +196,4 @@ def align_cluster(self, cluster: Atoms):
         principal_axes = pca.components_
         rotated_cluster = np.dot(cluster_centered, principal_axes.T)
         cluster.positions = rotated_cluster
-        return cluster
+        return cluster
```
Comment (mbalfakeih) [inline Disturber.py:199]: What a useful change!

File: GlobalOptimizer.py [hunk 2]
```diff
@@ -45,7 +45,6 @@ def run(self, maxIterations):
         self.setup()
 
         while self.currentIteration < maxIterations and not self.isConverged():
-            print(self.currentIteration)
```
Comment (mbalfakeih) [inline GlobalOptimizer.py:48]: This is very helpful!

File: Disturber.py (hunk 1 above)
Comment (Martin092) [reply Disturber.py:199]: Thank you I tried very hard

**Review by mbalfakeih — APPROVED** [review]
7.8/10 too much code
I'm really sorry but this pull requerwst just had too much code, I didn't like it. Next time try adding less code, I don't like code. I hope you can take this into account in the future.

I think you should apologize, here's an example of a helpful apology:
I made a severe and continuous lapse in my judgement, and I don’t expect to be forgiven. I’m simply here to apologise.

What we came across in the woods that day was obviously unplanned. The reactions you saw on tape were raw; they were unfiltered. None of us knew how to react or how to feel. I should have never posted the video. I should have put the cameras down and stopped recording what we were going through.

There's a lot of things I should have done differently but I didn't. And for that, from the bottom of my heart, I am sorry. I want to apologise to the internet. I want to apologise to anyone who has seen the video. I want to apologise to anyone who has been affected or touched by mental illness, or depression, or suicide. But most importantly I want to apologise to the victim and his family.

For my fans who are defending my actions, please don't. I don’t deserve to be defended. My goal with my content is always to entertain; to push the boundaries, to be all-inclusive. In the world live in, I share almost everything I do. The intent is never to be heartless, cruel, or malicious. Like I said I made a huge mistake. I don’t expect to be forgiven, I’m just here to apologise. I'm ashamed of myself. I’m disappointed in myself. And I promise to be better. I will be better. Thank you.

Kind regards,
Moham Ballfucky

**Pull request merged**